.github/workflows/java-sample-app-s3-deploy.yml

```diff
@@ -79,14 +83,75 @@ jobs:
         with:
-          role-to-assume: arn:aws:iam::${{ env.ACCOUNT_ID }}:role/${{ secrets.E2E_TEST_ROLE_ARN }}
+          role-to-assume: arn:aws:iam::${{ env.ACCOUNT_ID }}:role/${{ env.E2E_TEST_ROLE_NAME }}
           aws-region: ${{ matrix.aws-region }}
 
-      - name: Build Main Jar
-        working-directory: sample-apps/springboot-remote-service
-        run: gradle build
+      - name: Build and Upload Main Jar
+        working-directory: sample-apps/java/springboot-remote-service
+        run: |
+          gradle build -P javaVersion=11
+          aws s3api put-object --bucket aws-appsignals-sample-app-prod-${{ matrix.aws-region }} --body build/libs/springboot-remote-service-*-SNAPSHOT.jar --key java-remote-service-v11.jar
+
+  java-main:
+    strategy:
+      fail-fast: false
+      matrix:
+        java-version: [ '8', '17', '21', '22' ]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-java@v4
+        with:
+          distribution: 'temurin'
+          java-version: ${{ matrix.java-version }}
+          check-latest: true
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          role-to-assume: arn:aws:iam::${{ env.E2E_TEST_ACCOUNT_ID }}:role/${{ env.E2E_TEST_ROLE_NAME }}
+          aws-region: us-east-1
+
+      - name: Build and Upload Main Jar
+        working-directory: sample-apps/java/springboot-main-service
+        run: |
+          if [ "${{ matrix.java-version }}" = "22" ]; then
+            sed -i 's/id("org.springframework.boot")/id("org.springframework.boot") version "3.3.4"/' build.gradle.kts
+            cat build.gradle.kts
+          fi
+          
+          gradle build -P javaVersion=${{ matrix.java-version }}
+          aws s3api put-object --bucket aws-appsignals-sample-app-prod-us-east-1 --body ./build/libs/springboot-*-SNAPSHOT.jar --key java-main-service-v${{ matrix.java-version }}.jar
+
+  java-remote:
+    strategy:
+      fail-fast: false
+      matrix:
+        java-version: [ '8', '17', '21', '22' ]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-java@v4
+        with:
+          distribution: 'temurin'
+          java-version: ${{ matrix.java-version }}
+          check-latest: true
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          role-to-assume: arn:aws:iam::${{ env.E2E_TEST_ACCOUNT_ID }}:role/${{ env.E2E_TEST_ROLE_NAME }}
+          aws-region: us-east-1
+
+      - name: Build and Upload Remote Jar
+        working-directory: sample-apps/java/springboot-remote-service
+        run: |
+          if [ "${{ matrix.java-version }}" = "22" ]; then
+            sed -i 's/id("org.springframework.boot")/id("org.springframework.boot") version "3.3.4"/' build.gradle.kts
+            cat build.gradle.kts
+          fi
```

Comment:
Leave a comment as to why we need this only for 22

Reply:
Added